sql/core/src/main/scala/org/apache/spark/sql/execution/objects.scala

```diff
@@ -216,10 +217,39 @@ case class MapElementsExec(
     child.asInstanceOf[CodegenSupport].produce(ctx, this)
   }
 
+  private def getMethodType(dt: DataType, isOutput: Boolean): String = {
+    dt match {
+      case BooleanType if isOutput => "Z"
```

Comment:
IIUC, [this code](https://github.com/scala/scala/blob/v2.12.0/src/library/scala/Function1.scala#L32https://github.com/scala/scala/blob/v2.12.0/src/library/scala/Function1.scala#L32) specializes boolean type only for return type.